torchao/quantization/quant_api.py

```diff
@@ -755,6 +776,60 @@ def apply_float8_dynamic_activation_quant(weight: torch.Tensor):
     return _get_linear_subclass_inserter(apply_float8_dynamic_activation_quant)
 
 
+def float8_static_activation_float8_weight(
+    scale: torch.Tensor,
+    activation_dtype: torch.dtype = torch.float8_e4m3fn,
+    weight_dtype: torch.dtype = torch.float8_e4m3fn,
+    granularity: Optional[
+        Union[_fp8_granularities, Tuple[_fp8_granularities, _fp8_granularities]]
+    ] = None,
+    mm_config: Optional[Float8MMConfig] = None,
+):
+    """
+    Applies float8 static symmetric quantization to
+
+    Args:
+        scale (torch.Tensor): The scale tensor for activation quantization. TODO: does passing in 1 tensor scale make sense?
```

Comment:
This still exists,  is agood TODO. I want this flow to work with the insert_observers_ api

```Python
def is_weight_observer(mod: torch.Tensor, name: str) -> bool:
    return isinstance(mod, torch.nn.Linear) and isinstance(
        mod.weight, LinearActivationWeightObservedTensor
    )


def float8_static_activation_float8_weight_from_observers(
    lin_mod: torch.nn.Module,
):
    act_observer = lin_mod.weight.input_observer
    assert isinstance(
        act_observer, AffineQuantizedMinMaxObserver
    ), f"Float8 static quantization requires activation observer to be AffineQuantizedMinMaxObserver, got {type(act_observer)}"
    act_scale, act_zero_point = act_observer.calculate_qparams()
    quant_func = float8_static_activation_float8_weight(act_scale)
    # I dont like this next line
    lin_mod.weight = torch.nn.Parameter(lin_mod.weight.original_weight_tensor)
    return quant_func(lin_mod)

insert_observers_(m, acts_observer, weight_observer=None)
# calibrating / training
for _ in range(10):
    m(*example_inputs)

after_obs = m(*example_inputs)

# quantized linear represented as an nn.Linear with modified tensor subclass weights
# for both activation and weight quantization
quantize_(
    m,
    float8_static_activation_float8_weight_from_observers,
    filter_fn=is_weight_observer,
)
```

In which case it gets called in a "local" sense sense and 1 scale gets associated to 1 activation, but it is "possible" to use to quantize multiple linears with the same scale. I did find after sometesting though that this might be okay and is deff helpful for testing...

Reply:
what is the problem with 1 scale gets associated to 1 activation?

Reply:
This is what we want, but it is pretty easy to associate 1 scale with many activations:

`quantize_(model, float8_static_activation_float8_weight(scale=torch.tensor(1.0, device="cuda")))`
Which Is probably okay in some situations but idk

Reply:
you mean the same scale Tensor instance is used in multiple places? that seems fine?